File: backend/app/schemas/invitation.py
"""Pydantic schemas for invitation management."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, EmailStr


class CreateInviteRequest(BaseModel):
    """Request schema for creating an invitation."""

    email: EmailStr


class InviteResponse(BaseModel):
    """Response schema for a single invitation."""

    id: UUID
    email: str
    status: str
    created_at: datetime
    expires_at: datetime
    accepted_at: datetime | None = None


class InviteListResponse(BaseModel):
    """Paginated list of invitations."""

    items: list[InviteResponse]
    total: int
    page: int
    page_size: int


class InviteDetailResponse(BaseModel):
    """Detailed invitation response with inviter info."""

    id: UUID
    email: str
    status: str
    created_at: datetime
    expires_at: datetime
    accepted_at: datetime | None = None
    invited_by: UUID | None = None


class DuplicateInviteWarning(BaseModel):
    """Warning response for duplicate pending invitation."""

    message: str
    existing_invite_id: UUID

File: admin-frontend/src/types/invitation.ts
/**
 * TypeScript types for admin invitation management.
 * Mirrors backend schemas: invitation.py
 */

export interface Invitation {
  id: string;
  email: string;
  status: string;
  created_at: string;
  expires_at: string;
  accepted_at: string | null;
}

export interface InvitationListResponse {
  items: Invitation[];
  total: number;
  page: number;
  page_size: number;
}

export interface CreateInvitationRequest {
  email: string;
}

export interface InvitationListParams {
  page?: number;
  page_size?: number;
  status?: string | null;
}
